src/getting-started.md

```diff
@@ -0,0 +1,406 @@
+`rustc` has a 2-stage build.
+
+We have a special tool `./x.py` that drives this process. It is used for
+compiling the compiler, the standard libraries, and `rustdoc`. It is also used
+for driving CI and building the final release artifacts.
+
+### Building and Testing `rustc`
+
+For most contributions, you only need to build stage 1, which saves a lot of time.
+After updating `config.toml`, as mentioned above, you can use `./x.py`:
+
+```shell
+# Build the compiler (stage 1)
+./x.py build --stage 1
+
+# Subsequent builds
+./x.py build --stage 1 --keep-stage 1
+```
+
+This will take a while, especially the first time. Be wary of accidentally
+touching or formatting the compiler, as `./x.py` will try to recompile it.
+
+**NOTE**: The `--keep-stage 1` will _assume_ that the stage 0 standard library
+does not need to be rebuilt, which is usually true, which will save some time.
+However, if you are changing certain parts of the compiler, this may lead to
+weird errors. Feel free to ask on [zulip][z] if you are running into issues.
+
+To run the compiler's UI test suite (the bulk of the test suite):
+
+```
+# UI tests
+./x.py test --stage 1 src/test/ui [--keep-stage 1]
+```
+
+This will build the compiler first, if needed.
+
+This will be enough for most people. Notably, though, it mostly tests the
+compiler frontend, not codegen or debug info.  You can read more about the
+different test suites [in this chapter][testing].
+
+[testing]: https://rustc-dev-guide.rust-lang.org/tests/intro.html
+
+If you only want to check that the compiler builds (without actually building
+it) you can run the following:
+
+```shell
+./x.py check
+```
+
+To format the code:
+
+```shell
+# Actually format
+./x.py fmt
+
+# Just check formatting, exit with error
+./x.py fmt --check
+```
+
+You can use `RUSTC_LOG=XXX` to get debug logging. [Read more here][logging].
+
+[logging]: ./compiler-debugging.html#getting-logging-output
+
+### Building and Testing `std`/`core`/`alloc`/`test`/`proc_macro`/etc.
+
+To contribute to `libstd`, you don't need to build the compiler unless you are
+planning to use a recently added nightly feature. Instead, you can just build
+stage 0.
+
+```sh
+./x.py build --stage 0 src/libstd
+```
+
+TODO: how to test?
+
+### Building and Testing `rustdoc`
+
+`rustdoc` uses `rustc` internals (and, of course, the standard library), so you
+will have to build the compiler and `std` once before you can build `rustdoc`.
+
+The following command will build all of them. Stage 1 should be sufficient,
+even though the release version will use the full 2-stage build.
+
+```sh
+# First build
+./x.py build --stage 1 src/tools/rustdoc
+
+# Subsequent builds
+./x.py build --stage 1 --keep-stage 1 src/tools/rustdoc
+```
+
+You can also use `./x.py check` here to do a fast check build.
+
+TODO: how to test?
+
+### Contributing code to other Rust projects
+
+There are a bunch of other projects that one can contribute too outside of the
```

Comment:
```suggestion
There are a bunch of other projects that you can contribute to outside of the
```

I suggest “you” because it’s what you use in the rest of the page